### PR TITLE
Pyupgrade notebooks

### PR DESCRIPTION
**Pull request by MarcoGorelli**
cc @aseyboldt 

This changes loads of files, but in most of them the diff is just
```diff
-    "print(\"Running on PyMC3 v{}\".format(pm.__version__))"
+    "print(f\"Running on PyMC3 v{pm.__version__}\")"
```

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,18 @@
...
+isort = 1
+black = 1
+pyupgrade = 1
+
+[tool.nbqa.addopts]
+isort = ["--treat-comment-as-code", "# %%"]
+pyupgrade = ["--py36-plus"]
+
+[tool.nbqa.ignore_cells]
+black = "%%cython"
```
Comment (AlexAndorra) [inline pyproject.toml]: Aren't those the same changes as in #4135 ?

File: .nbqa.ini [hunk 2]
```diff
@@ -1,4 +0,0 @@
-[isort]
-config = setup.cfg
-mutate = 1
-addopts = --treat-comment-as-code '# %%%%'
```
Comment (AlexAndorra) [inline .nbqa.ini]: Aren't those the same changes as in #4135 ?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -3,14 +3,20 @@ repos:
   rev: v3.2.0
   hooks:
     -   id: end-of-file-fixer
+    -   id: check-toml
 - repo: https://github.com/nbQA-dev/nbQA
-  rev: 0.1.29
+  rev: 0.2.0
```
Comment (AlexAndorra) [inline .pre-commit-config.yaml]: Aren't those the same changes as in #4135 ?

File: docs/source/notebooks/conditional-autoregressive-model.ipynb [hunk 4]
```diff
@@ -3110,7 +3110,7 @@
     "        n, m = W.shape\n",
     "        self.n = n\n",
     "        self.median = self.mode = self.mean = 0\n",
-    "        super(Sparse_CAR, self).__init__(*args, **kwargs)\n",
+    "        super().__init__(*args, **kwargs)\n",
```
Comment (AlexAndorra) [inline docs/source/notebooks/conditional-autoregressive-model.ipynb:3113]: Does this still work without the arguments passed in the old versions?

**Review by AlexAndorra — CHANGES_REQUESTED** [review]
Thanks a lot @MarcoGorelli ! I just had some questions below, as it seems to me that you picked up changes from #4135 in this PR, but maybe I'm mistaken.
Also, I just merged a PR that modified the GLM-Poisson NB, so you need to merge master into this branch, sorry about 😬

File: docs/source/notebooks/conditional-autoregressive-model.ipynb (hunk 4 above)
Comment (MarcoGorelli) [reply docs/source/notebooks/conditional-autoregressive-model.ipynb:3113]: I don't know if it works with versions of Python older than 3.6, although 3.6 is the oldest version currently supported by PyMC3 and this notebook lists Python 3.8 as the version it was run on so my thinking was that it was OK

File: .pre-commit-config.yaml (hunk 3 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml]: Yes - if want to accept #4135 then I can merge upstream/master, I just needed to upgrade the `nbqa` version in order to be able to set configs in `pyproject.toml`

File: docs/source/notebooks/conditional-autoregressive-model.ipynb (hunk 4 above)
Comment (AlexAndorra) [reply docs/source/notebooks/conditional-autoregressive-model.ipynb:3113]: Ow ok, didn't know that not passing the args to `super` worked now!

File: .pre-commit-config.yaml (hunk 3 above)
Comment (AlexAndorra) [reply .pre-commit-config.yaml]: Sounds good, just wanted to make sure it was on your radar 👍

**Comment by MarcoGorelli** [comment]
Thanks @AlexAndorra - have updated

**Review by AlexAndorra — APPROVED** [review]
Perfect, thanks @MarcoGorelli !

**Pull request merged**